### ISSUE TITLE
Add Lind-WASM Test Cases with build system and test automation

### ISSUE THREAD

**Issue by yzhang71** [Lind-Project/lind-wasm#11]
So for Lind, we have the test cases for each system call available here: https://github.com/Lind-Project/lind_project/tree/main/tests/test_cases. We want to handle the system calls in Lind-WASM as well. The task is to create Lind-WASM test cases for each system call, similar to the approach used in https://github.com/Lind-Project/lind_project/blob/main/tests/test_cases/suite.sh. If a system call is not supported, it needs to be added to the rustposix.

**Comment by rennergade** [comment]
Now that we have the monorepo, @robinyuan1002 can we port these tests in the "tests" folder. Let's also set up a build script to build all of them and a run script to test all of them against native like we had in rustposix.

**Comment by rennergade** [comment]
The next step here is to generate all the expected outputs.

@arnabdas1999 can you compile all of the c file tests for native linux using gcc in this folder: https://github.com/Lind-Project/lind-wasm/tree/main/tests/unit-tests and redirect the output to a *.expected file

ie
```
gcc hello.c -o hello
./hello > hello.expected
```

Once you have those can you PR them by adding them in a /expected subfolder for each category.

**Comment by rennergade** [comment]
@ansalma6 is going to take over the above task. @yashaswi2000 can you help walk him through this?

**Comment by rennergade** [comment]
We're getting closer to getting this up an integrated with CI

@qianxichen233 @Yaxuan-w here's the current report of whats succeeding and failing on main. 

[report.json](https://github.com/user-attachments/files/18758039/report.json)

@ansalma6 @m-hemmings are going to work on getting this integrated into CI and have the output be easier to read.

**Comment by Yaxuan-w** [comment]
> current report of whats succeeding and failing on main.

I'm a bit confused by the report... I could successfully run `rename.c` this morning with main branch. The error msg on report is 

```txt
"/home/lind/lind-wasm/tests/unit-tests/file_tests/deterministic/rename.c": {
    "status": "Failure",
    "error_type": "Failure_GCC_running",
    "output": "ERROR WITH STAT CALL ON OLD NAME: No such file or directory\n"
},
```

which I think is related to test setup? For example: creating test files in `src/RawPOSIX/tmp` before testing

**Comment by qianxichen233** [comment]
since vmmap has been merged, can we run the test again with the latest main branch? Vmmap is pretty big update so probably the test report could be much different

**Comment by rennergade** [comment]
@ansalma6 could you merge the current main into your PR and run it again? Woulld be great it you could drop the results in the new html format in here.

**Comment by ansalma6** [comment]
@rennergade I've attached the current report using the latest main branch.
[Report.zip](https://github.com/user-attachments/files/18799788/Report.zip)

@Yaxuan-w 
rename.c failed because it requires a file testfiles/renamefile.txt. That is missing in the repo, so the test case failed. 

It is possible to make the native compilation and running success by creating the required file and changing the working directory to the location where the testfiles are(I tested by using os.chdir in the python script before running the gcc binary) but the wasm runs still fails beacuse of the folder structure.

**Comment by Yaxuan-w** [comment]
> but the wasm runs still fails beacuse of the folder structure.

Could you post the error msg here?

**Comment by ansalma6** [comment]
I'm running in the docker container. 

I created a folder testfiles and renamefile.txt in it at the location where the rename.c is. 

Now when I run "./lindtool.sh compile_test tests/unit-tests/file_tests/deterministic/rename", the compilation works and the wasm binaries are created 

But when I run "./lindtool.sh run tests/unit-tests/file_tests/deterministic/rename", I get the output 
"/home/lind/lind-wasm/src/wasmtime/target/debug/wasmtime run --allow-precompiled --wasi threads=y --wasi preview2=n tests/unit-tests/file_tests/deterministic/rename.cwasm 
ERROR WITH RENAME: No such file or directory". 

Then, I created the testfiles and renamefile.txt at the same folder as lindtool.sh and got the same output

**Comment by ansalma6** [comment]
I was able to get it running when the testfiles/renamefile.txt is present in src/RawPOSIX/tmp folder. Is that where the required testfiles are supposed to be?

**Comment by Yaxuan-w** [comment]
> I was able to get it running when the testfiles/renamefile.txt is present in src/RawPOSIX/tmp folder. Is that where the required testfiles are supposed to be?
> 

yes. `src/RawPOSIX/tmp` is the root dir for lind-wasm fs perspective

**Comment by qianxichen233** [comment]
There is a directory for files used by test you should copy to lind filesystem: https://github.com/Lind-Project/lind_project/tree/3bdc390a149b381386a8f97e166d5fa6769dd45d/tests/test_cases/testfiles

I am not sure if this directory is removed in lind-wasm repo, but this should be copied into src/RawPOSIX/tmp before running the test. It would probably be required to copy this directory to kernel root folder to be able to run native test as well

**Comment by rennergade** [comment]
We can finally close this issue! Thanks @ansalma6 and @m-hemmings
